drivers/fuel_gauge/sbs_gauge/sbs_gauge.h

```diff
@@ -47,8 +47,38 @@
 
 #define SBS_GAUGE_DELAY                     1000
 
+/*
+ * Nearly all cutoff payloads are actually a singular value that must be written twice to the fuel
+ * gauge. For the case where it's a singular value that must only be written to the fuel gauge only
+ * once, retransmitting the duplicate write has no significant negative consequences.
+ *
+ * Why not devicetree: Finding the maximum length of all the battery cutoff payloads in a devicetree
+ * at compile-time would require labyrinthine amount of macro-batics.
+ *
+ * Why not compute at runtime: It's not worth the memory given having more than a single fuel gauge
+ * is rare, and most will have a payload size of 2.
+ *
+ * This is validated as a BUILD_ASSERT in the driver.
+ */
+#define SBS_GAUGE_CUTOFF_PAYLOAD_MAX_SIZE 2
+
+#define _SBS_GAUGE_INST_CUTOFF_SUPPORTED(inst)                                                     \
+	DT_PROP_OR(DT_DRV_INST(inst), battery_cutoff_support, false)
+#define SBS_GAUGE_CUTOFF_SUPPORT()                                                                 \
+	(false || DT_INST_FOREACH_STATUS_OKAY(_SBS_GAUGE_INST_CUTOFF_SUPPORTED))
```

Comment:
Interesting approach :-)

How about instead of this you split the cutoff fields in a separate config struct, only instantiate it if the specific instance has the property and then leave the handler there all the time (it only takes few bytes of flash) and return an error instantly if this structure is not set? (meaning the battery does not support it)

Or I guess you could rip off the multi instance support if we don't need it, though that'd be sad, I really miss my X250 :_(

Reply:
Oh you already have `cutoff_support`, so yeah you could replace that with a struct pointer, if it's null you don't have cutoff support, if it's set you do and the data is there.

Reply:
Sgtm, having the extra struct is easier to manage, gives grouping to the associated properties.

Reply:
@fabiobaltieri But allocating a different struct makes no change to needing `SBS_GAUGE_CUTOFF_SUPPORT()` unless I'm missing something?

You're talking about just collecting constant parameters as a separate struct, right?

Reply:
You should be able to drop `SBS_GAUGE_CUTOFF_SUPPORT()` entirely if you only allocate the extra structure if `battery_cutoff_support` is there, then `sbs_gauge_do_battery_cutoff` is always compiled in but validates that the structure is not there and returns `-ENOSUP` in that case.

Reply:
Oh that's certainly cleaner. I'm totally happy to do that if we don't care about `sbs_gauge_do_battery_cutoff` always being compiled in and nominally increasing the code size.

However, I think it's more broadly suitable if we keep it compiled out. Battery cutoff is *NOT* standard SBS behavior so many folks might be willing to use this driver without wanting the battery cutoff feature.

I'll do that then. We can always revert it back to using the helper macro if requested :)

Reply:
You can always have a Kconfig option to compile that out if one really wants to save those few bytes.